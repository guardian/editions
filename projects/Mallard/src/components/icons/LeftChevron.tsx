import React from 'react'
import { Text, StyleSheet } from 'react-native'
import { getFont } from 'src/theme/typography'
import { color } from 'src/theme/color'

const LeftChevron = ({
    fill = color.palette.neutral[20],
}: {
    fill?: string
}) => {
    const styles = StyleSheet.create({
        icon: {
            ...getFont('icon', 1),
            color: fill,
        },
    })

    return <Text style={styles.icon}>{'\uE00A'}</Text>
}

export { LeftChevron }
