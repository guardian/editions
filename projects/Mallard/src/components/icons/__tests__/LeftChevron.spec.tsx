import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { LeftChevron } from '../LeftChevron';

describe('LeftChevron', () => {
	it('should display a LeftChevron icon using the icon font', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<LeftChevron />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
