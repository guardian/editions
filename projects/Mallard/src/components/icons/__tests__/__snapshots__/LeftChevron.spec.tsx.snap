// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftChevron should display a LeftChevron icon using the icon font 1`] = `
<Text
  style={
    Object {
      "color": "#333333",
      "fontFamily": "GuardianIcons-Regular",
      "fontSize": 20,
      "lineHeight": 20,
    }
  }
>
  î€Š
</Text>
`;
